File: Backend/src/main/java/com/example/easynotes/model/KhachHang.java
package com.example.easynotes.model;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.Nationalized;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.LastModifiedDate;
import org.springframework.data.jpa.domain.support.AuditingEntityListener;

import javax.persistence.*;
import javax.validation.constraints.NotBlank;
import java.io.Serializable;
import java.util.Date;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
@EntityListeners(AuditingEntityListener.class)
@JsonIgnoreProperties(value = {"createdAt", "updatedAt"}, allowGetters = true)
@Entity
public class KhachHang implements Serializable {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long maKhachHang;

    @Nationalized
    @NotBlank
    private String tenKhachHang;

    @Embedded
    private DiaChi diaChiNhanHang;

    @OneToOne
    @JoinColumn(name = "tenTaiKhoan")
    private TaiKhoan taiKhoan;

    @NotBlank
    private String soDienThoai;

    @Column(nullable = false, updatable = false)
    @Temporal(TemporalType.TIMESTAMP)
    @CreatedDate
    private Date createdAt;

    @Column(nullable = false)
    @Temporal(TemporalType.TIMESTAMP)
    @LastModifiedDate
    private Date updatedAt;
}

File: App/go_quick_app/lib/models/khach_hang.dart
import 'package:go_quick_app/models/dia_chi.dart';
import 'package:go_quick_app/models/tai_khoan.dart';

class KhachHang {
  KhachHang({
    this.maKhachHang,
    this.tenKhachHang,
    this.diaChiNhanHang,
    this.taiKhoan,
    this.soDienThoai,
    this.createdAt,
    this.updatedAt,
  });

  int? maKhachHang;
  String? tenKhachHang;
  DiaChi? diaChiNhanHang;
  TaiKhoan? taiKhoan;
  String? soDienThoai;
  DateTime? createdAt;
  DateTime? updatedAt;

  factory KhachHang.fromJson(Map<String, dynamic> json) => KhachHang(
        maKhachHang: json["maKhachHang"],
        tenKhachHang: json["tenKhachHang"],
        diaChiNhanHang: DiaChi.fromJson(json["diaChiNhanHang"]),
        taiKhoan: TaiKhoan.fromJson(json["taiKhoan"]),
        soDienThoai: json["soDienThoai"],
        createdAt: json["createdAt"] != null
            ? DateTime.parse(json["createdAt"])
            : null,
        updatedAt: json["updatedAt"] != null
            ? DateTime.parse(json["updatedAt"])
            : null,
      );

  Map<String, dynamic> toJson() => {
        "maKhachHang": maKhachHang,
        "tenKhachHang": tenKhachHang,
        "diaChiNhanHang": diaChiNhanHang?.toJson(),
        "taiKhoan": taiKhoan?.toJson(),
        "soDienThoai": soDienThoai,
      };
}
